experimental/schoology/export-assignment-data.py

```diff
@@ -0,0 +1,269 @@
+            section_list += [
+                c
+                for c in c.get_sections()
+                for gp in self.grading_periods_lookup
+                if gp in c.grading_periods
+            ]
+
+        # Reshape for our desired output
+        def dto_to_section(section):
+            return {
+                "id": section.id,
+                "course_id": section.course_id,
+                "school_id": section.school_id,
+                "course_title": section.course_title,
+                "course_code": section.course_code,
+                "section_title": section.section_title,
+                "grading_period_ids": section.grading_periods,
+                "grading_period_names": [
+                    self.grading_periods_lookup[gp] for gp in section.grading_periods
+                ],
+                "enrollments": {
+                    e["id"]: e["uid"]
+                    for e in self.sc_client.get_enrollments(section.id)
+                },
+            }
+
+        return [dto_to_section(s) for s in section_list]
+
+    def _full_name(self, student):
+        names = [student["name_first"], student["name_middle"], student["name_last"]]
+        return " ".join(x for x in names if x)
+
+    def _find_student(self, students, uid):
+        return next(s for s in students if s["uid"] == uid)
+
+    def _get_assignments(self, students, sections):
+        def dto_to_assignment(section, assignment):
+            a = {
+                "id": assignment.id,
+                "section_id": section["id"],
+                "section_title": section["section_title"],
+                "title": assignment.title,
+                "due": assignment.due,
+                "max_points": assignment.max_points,
+                "is_final": assignment.is_final,
+                "type": assignment.type,
+                "grade_item_id": assignment.grade_item_id,
+                "grading_period_id": assignment.grading_period,
+                "grading_period_name": None,
+                "assignee_ids": assignment.assignees,
+                "assignee_names": [],
+            }
+
+            if int(assignment.grading_period) in self.grading_periods_lookup:
+                a["grading_period_name"] = self.grading_periods_lookup[
+                    int(assignment.grading_period)
+                ]
+
+            if assignment.assignees:
+                # Need to look up students vai the section['enrollments'], which
+                # is of the form {section_id : student_uid}.
+                enrollments = section["enrollments"]
+                assignees = [
+                    enrollments[str(i)]
+                    for i in assignment.assignees
+                    if str(i) in enrollments
+                ]
+                a["assignee_names"] = [
+                    self._full_name(self._find_student(students, a)) for a in assignees
+                ]

```

Comment:
Typically an assignment is given to an entire class, but in Schoology you can also give it to individuals. This flattens the list into a single column.